### ISSUE TITLE
indentation region after `return` keyword is not recognized

### ISSUE THREAD

**Issue by unkarjedy** [lampepfl/dotty#11802]
## Compiler version
3.0.0-RC2-bin-20210317-758782a-NIGHTLY
## Minimized code

```Scala
def bar2: Int = {
  println(1)
  println(2)
  return
    var x = 1
    var y = 1
    x + y
}

bar2
```

## Output
```
return              
^^^^^^              
Found:    Unit      
Required: Int       
```
![image](https://user-images.githubusercontent.com/3989292/111612357-72a03400-87ee-11eb-86f8-c13f5ac547b2.png)

## Expectation
According to: 
http://dotty.epfl.ch/docs/reference/other-new-features/indentation.html#optional-braces
>... an indentation region can start ... after one of the following tokens:
> ... **return** ...

So the code should compile, or the reference should be adjusted.

**Comment by unkarjedy** [comment]
If you extract the return result to a variable it works:

![image](https://user-images.githubusercontent.com/3989292/111612940-0bcf4a80-87ef-11eb-8621-9180001f7df5.png)

Though the output is badly-sorted: `def bar2: Int` should go before function call (REPL issue?)

**Comment by abgruszecki** [comment]
@unkarjedy did the output look different in Scala2? I don't think there's much we can do about that particular part, considering that one can define multiple vals in one REPL input...

**Comment by som-snytt** [comment]
I recently revisited a closed scala 2 ticket about order of REPL output, which at least demonstrates that there are differing expectations. https://github.com/scala/bug/issues/10893

**Comment by odersky** [comment]
This will be fixed by #11752
